integration-tests/tests/run-analysis-azure.js

```diff
@@ -0,0 +1,88 @@
+} = require('../utils/integration-utils');
+const { registerTest } = require('../utils/jest-utils');
+const { withUserToken } = require('../utils/terra-sa-utils');
+
+const notebookName = `test-notebook-${uuid.v4()}`;
+
+const testRunAnalysisAzure = _.flowRight(
+  withUserToken,
+  withAzureWorkspace
+)(async ({ billingProject, workspaceName, page, testUrl, token }) => {
+  await gotoAnalysisTab(page, token, testUrl, workspaceName);
+
+  // Create analysis file
+  await click(page, clickable({ textContains: 'Start' }));
+  await findElement(page, getAnimatedDrawer('Select an application'));
+  await click(page, image({ text: 'Create new notebook' }));
+  await fillIn(page, input({ placeholder: 'Enter a name' }), notebookName);
+  await noSpinnersAfter(page, { action: () => click(page, clickable({ text: 'Create Analysis' })) });
+
+  // Dismiss the create env modal for now
+  await noSpinnersAfter(page, {
+    action: () => findText(page, 'A cloud environment consists of application configuration, cloud compute and persistent disk(s).'),
+  });
+  await click(page, clickable({ textContains: 'Close' }));
+  await waitForNoModal(page);
+
+  // Navigate to analysis launcher
+  await click(page, `//*[@title="${notebookName}.ipynb"]`);
+  await dismissNotifications(page);
+  await findText(page, 'PREVIEW (READ-ONLY)');
+
+  // Attempt to open analysis; create a cloud env
+  await click(page, clickable({ textContains: 'Open' }));
+  await findText(page, 'Azure Cloud Environment');
+  await click(page, clickable({ textContains: 'Create' }));
+  await waitForNoModal(page);
+
+  // Wait for env to begin creating
+  await findElement(page, clickable({ textContains: 'JupyterLab Environment' }));
+  await findElement(page, clickable({ textContains: 'Creating' }));
+
+  // Wait for env to finish creating
+  await Promise.race([
+    findElement(page, clickable({ textContains: 'Running' }), { timeout: Millis.ofMinutes(15) }),
+    findErrorPopup(page, { timeout: Millis.ofMinutes(15) }),
+  ]);
+  await click(page, clickable({ textContains: 'Open' }));
+
+  // Find the iframe and wait until the Jupyter kernel is ready
+  const frame = await findIframe(page, '//iframe[@title="Interactive JupyterLab iframe"]', { timeout: Millis.ofMinutes(2) });
+  await findText(frame, 'Kernel status: Idle', { timeout: Millis.ofMinutes(2) });
+
+  // Run a command
+  await fillIn(frame, '//textarea', 'print(123456789099876543210990+9876543219)');
+  await click(frame, '//button[starts-with(@title, "Run the selected cells and advance")]');
+  await findText(frame, '123456789099886419754209');
+
+  // Save notebook to avoid "unsaved changes" modal when test tear-down tries to close the window
+  await click(frame, '//button[starts-with(@title, "Save and create checkpoint")]');
+
+  // Cleanup
+  await deleteRuntimesV2({ page, billingProject, workspaceName });
+});
+
+registerTest({
+  name: 'run-analysis-azure',
+  fn: testRunAnalysisAzure,
+  timeout: Millis.ofMinutes(20),
```

Comment:
You will need to register that your test only runs in dev/staging by specifing
`targetEnvironments: ['dev', 'staging']`

Reply:
oh ho ho ho the hidden knowledge! thank you :)

Reply:
Oh, no you pushed something up! Please don't push another commit before 4 so we can see how long the test takes (and if everything passes). When a commit is pushed, it kills the running tests.

Reply:
Np! That should be the last thing

Reply:
Unfortunately I don't think it made it past the first line of the test (await gotoAnalysisTab(page, token, testUrl, workspaceName)). I'm going to hold off on reviewing until you get it passing reliably. 

I'll rerun the test though to see if the current issue is a flake, and perhaps we can get some timing information.

Reply:
Screenshot: https://circleci-tasks-prod.s3.us-east-1.amazonaws.com/storage/artifacts/1c1bde7e-0f01-4eef-b2bc-01f4b12ed651/4cccc75d-08a4-41b6-a1ba-c3fbdd237675/2/results/failure-screenshots/failure-1699389914239-run-analysis-azure.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQVFQINEOBNNIXZ4O%2F20231107%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20231107T205121Z&X-Amz-Expires=60&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEHUaCXVzLWVhc3QtMSJHMEUCIQDR5QJwVKMkrvw88dZrzUfG5tShdfsIm%2B%2FBZH0Z6nytmQIgLr2unViJ7g0i%2FNN07noOyh5lRtTOJkNXC52aovd6BtcqtAIIrv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARADGgwwNDU0NjY4MDY1NTYiDDwfs6DSpxLw2WjrDCqIAuCv8pdZty7wtQE4kKxbHYo9Be7NRvzGQcA9E6g3KX7Ls6UvlqWR76Vju1ly2yZ5KpjS%2BuLwMp8UAiskUtZOXi5V%2Fo0wqJ6%2FNg27GmPWEgPHGFJY8DQsXI4y%2BC0ku5gD4v6bM4wOG%2BHxRP%2Fpno4fUf2M%2BBZX1HegF1FO%2F3jIL61os8mouRT3Tjua%2FGNECjSIqxeEQzTvhWSqi0uzktoSriL%2BOlKbFc9Lz9aLZhi0ad53EVNxrnpausmjEG3dO23%2F4%2FoFq5OpPNMCL68sochFJdzp8LTzIPJopjA8cxEhApEOXEdaRob%2FmSrEoOhMgvmrHa%2BLcOPzp8zKt%2BgVR5BG8wkEthCyOrznOTCawKqqBjqdATnsgbdpujcxrONyUVbKRFbD9zdXYOlgc49EXNs0HS6rtzSuKBcHvpgqhO1PE0pSwBpW7%2ByV6RMe3J959%2Ffb%2BXUck%2B0P5oUDIrar0u2crYElpkuJ59aaQS5i%2BVkx8fuKPSz8dl7A3e%2B90g5C%2FuiIhl08bhjyBW63Yng1T7NYhqJCmrlawrr%2Fs2QiQu6sOMrnxSWC68ibIMyzSv%2BqMjA%3D&X-Amz-SignedHeaders=host&x-id=GetObject&X-Amz-Signature=f5b1441a15c26793ebdf6309e35ccffc1de8b5da00556ebb2c3af4e015f2ea44

Reply:
This isn't a flake, it's a vexing issue that just started happening this morning. https://broadworkbench.atlassian.net/browse/IA-4682